ddl/callback.go

```diff
@@ -77,3 +86,132 @@ func (c *BaseCallback) OnJobUpdated(job *model.Job) {
 func (c *BaseCallback) OnWatched(ctx context.Context) {
 	// Nothing to do.
 }
+
+// SetDomain implements Callback.SetDomain interface.
+func (c *BaseCallback) SetDomain(do DomainReloader) {
+	// Nothing to do.
+}
+
+// DomainReloader is used to avoid import loop.
+type DomainReloader interface {
+	Reload() error
+}
+
+// ****************************** Start of Customized DDL Callback Instance ****************************************
+
+// DefaultCallback is the default callback that TiDB will use.
+type DefaultCallback struct {
+	*BaseCallback
+	do DomainReloader
+}
+
+// OnChanged overrides ddl Callback interface.
+func (c *DefaultCallback) OnChanged(err error) error {
+	if err != nil {
+		return err
+	}
+	logutil.BgLogger().Info("performing DDL change, must reload")
+
+	err = c.do.Reload()
+	if err != nil {
+		logutil.BgLogger().Error("performing DDL change failed", zap.Error(err))
+	}
+
+	return nil
+}
+
+// OnSchemaStateChanged overrides the ddl Callback interface.
+func (c *DefaultCallback) OnSchemaStateChanged() {
+	err := c.do.Reload()
+	if err != nil {
+		logutil.BgLogger().Error("domain callback failed on schema state changed", zap.Error(err))
+	}
+}
+
+// SetDomain is used to assign the domain.
+func (c *DefaultCallback) SetDomain(do DomainReloader) {
+	c.do = do
+}
+
+func newDefaultCallBack(do DomainReloader) Callback {
+	return &DefaultCallback{do: do}
+}
+
+// ****************************** End of Default DDL Callback Instance *********************************************
+
+// ****************************** Start of CTC DDL Callback Instance ***********************************************
+
+// CtcCallback is the customized callback that TiDB will use.
```

Comment:
```suggestion
// ctcCallback is the customized callback that TiDB will use.
```

And we'd better add a comment for `ctc` here.